docs/agentic-development/roadmap/openshell/sandbox-setup-guide.md

```
@@ -0,0 +1,260 @@
+## Step 3: Upload GCP credentials
+
+The `gcp-adc` provider injects credentials via openshell's proxy placeholder system, which doesn't work for ADC file-based auth (the proxy can intercept HTTP headers but not local file reads). Upload the ADC file directly:
+
+```bash
+openshell sandbox upload my-project \
+  "$HOME/.config/gcloud/application_default_credentials.json" \
+  /sandbox/adc.json
+```
+
+> **Note:** `openshell sandbox upload` treats the destination as a directory and places the source file inside it. So the actual file path inside the sandbox will be `/sandbox/adc.json/application_default_credentials.json`.
+
```

markdownlint-cli2 (0.22.0)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)